src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1249,7 +1249,7 @@ static void UpdateCurrentState(EvmState state, int pc, long gas, int stackHead)
                         Address address = stack.PopAddress();
                         if (!ChargeAccountAccessGas(ref gasAvailable, vmState, address, spec)) goto OutOfGas;
 
-                        if (programCounter < code.Length)
+                        if (!traceOpcodes && programCounter < code.Length)
```

Comment:
If we had it extracted, it would be like

```csharp
if (TryFindOptimization(nextInstruction, stack, ...)){

}
```

😛

Reply:
3 examples to generalise the pattern 😉